### ISSUE TITLE
Generating with mbart-large-cc25: which special tokens to use?

### ISSUE THREAD

**Issue by sshleifer** [pytorch/fairseq#2258]
I am trying to figure out how to use the posted cc25 model for generation. 
When I port the model to huggingface/transformers, I can perfectly copy the state dict. However, when I try to translate, using the same special token setup as `mbart-large-en-ro`, my english-romanian translator just generates english text.
Is this special tokens setup correct:
`english/src`: end src sentence with [lang_id, eos], so [250004, 2]

I tried to start decoding with the romanian langid, 250020, and got back english.
If I start decoding without a prefix token, I get back gibberish.

Alternatively, if anyone has a working `fairseq-generate` command that uses cc25 to generate (for any language pair), I could get to the bottom of which special tokens to use by setting breakpoints.

[here](https://gist.github.com/sshleifer/2e5c15d3ee428c0b7fed6407e3d9b419) is a broken attempt at a `fairseq-generate` command, the solutions proposed in #1758 do not seem to fix it. 

Thanks in advance!

**Comment by kaushal0494** [comment]
@myleott I am trying to use the mBART generative interface to generated output in the Hindi language (from English). I fine-tuning the model with English-Hindi parallel dataset and results are good. 

For my ongoing work, what I want is: "can we generate the Hindi output such that the order of sentences should not be change after generation? (i.e., English sentence order and generated Hindi sentence order should be same, there should not be any shuffling)" How can we achieve the same? Waiting for your response. Thank you!
